src/dune_rpc_server/dune_rpc_server.mli

```diff
@@ -1,6 +1,34 @@
 open Stdune
 open Dune_rpc_private
 
+module Subscription : sig
+  (** Represents a live subscription for values of type ['a] *)
+
+  type 'a t
+
+  type packed = E : _ t -> packed
+
+  val compare_packed : packed -> packed -> Ordering.t
+
+  val to_dyn : _ t -> Dyn.t
+
+  val compare : 'a t -> 'a t -> Ordering.t
+
+  (** Send the next value in the subscription *)
+  val update : 'a t -> 'a -> unit Fiber.t
+
+  (** Signal that this subscription is terminated. It is an error to call
+      [update] after [finish]. Subsequent calls to [finish ] do nothing. *)
+  val finish : _ t -> unit Fiber.t
+
+  (** Checks if the subscription is still active. A subscription can be
+      terminated either by calling [finish] or waiting for the client to
+      terminate it. It is illegal to call [update] if [is_alive] returns [false] *)
```

Comment:
Meant to write `active`, not `is_alive`? Also, this requirement that you can't call `update` if the client terminates the subscription is weird. Are you supposed to guard every call to `update` by the `active` check? If so, why don't we combine them into one function?